docs/source/tutorials/arch_modeling/spypads_tutorial.rst

```diff
@@ -0,0 +1,333 @@
+
+.. code-block:: bash
+
+   gtkwave openfpga_flow/tasks/fpga_verilog/spypad/latest/k6_frac_N10_tileable_adder_register_scan_chain_depop50_spypad_40nm/and2/MIN_ROUTE_CHAN_WIDTH/and2_formal.vcd &
+
+.. note:: Information on GTKWave can be found on our documentation page located here: :ref:`from_verilog_to_verification`
+
+The waveforms will appear similar to :numref:`fig_spypad_waves`
+
+.. _fig_spypad_waves:
+
+.. figure:: ./figures/spypad_waveforms.png
+   :scale: 100%
+
+   Spypad Waveforms
+
+Building Spypads
+~~~~~~~~~~~~~~~~
+
+We will modify the `k6_frac_N10_adder_chain_40nm_openfpga.xml <https://github.com/lnis-uofu/OpenFPGA/blob/tutorials/openfpga_flow/openfpga_arch/k6_frac_N10_adder_chain_40nm_openfpga.xml>`_ file found in OpenFPGA to expose the **sumout** output from the **ADDF** module. We can start modifying
+the file by running the following command:
+
+.. code-block:: bash
+
+   emacs openfpga_flow/openfpga_arch/k6_frac_N10_adder_chain_40nm_openfpga.xml
+
+Replace **LINE214** with the following:
+
+.. code-block:: xml
+
+   <port type="output" prefix="sumout" lib_name="SUM" size="1" is_global=”true” is_io=”true”/>
+
+**sumout** is now a global output. **sumout** will show up in the ``fpga_top.v`` file and will have waveforms in GTKWave if we run the **full testbench**. To run the 
+**full testbench**, we have to modify the ``hard_adder`` configuration file:
+
+.. code-block:: bash
+
+   emacs openfpga_flow/tasks/fpga_verilog/adder/hard_adder/config/task.conf
+
+Comment out the last line of the file to run the **full testbench**:
+
+.. code-block:: python
+
+   #vpr_fpga_verilog_formal_verification_top_netlist=
+
+We now run the task to see our changes:
+
+.. code-block:: bash
+
+   python3 openfpga_flow/scripts/run_fpga_task.py fpga_verilog/adder/hard_adder --debug --show_thread_logs
+
+We can view the global ports in ``fpga_top.v`` by running the following command:
+
+.. code-block:: bash
+
+   emacs openfpga_flow/tasks/fpga_verilog/adder/hard_adder/run064/k6_frac_N10_tileable_adder_chain_40nm/and2/MIN_ROUTE_CHAN_WIDTH/SRC/fpga_top.v
+
+The ``fpga_top.v`` should have the following in its module definition:
+
+.. code-block:: verilog
+
+   module fpga_top(pReset,
+                prog_clk,
+                set,
+                reset,
+                clk,
+                gfpga_pad_ADDF_sumout,
+                gfpga_pad_GPIO_PAD,
+                ccff_head,
+                ccff_tail);
+   //----- GLOBAL PORTS -----
+   input [0:0] pReset;
+   //----- GLOBAL PORTS -----
+   input [0:0] prog_clk;
+   //----- GLOBAL PORTS -----
+   input [0:0] set;
+   //----- GLOBAL PORTS -----
+   input [0:0] reset;
+   //----- GLOBAL PORTS -----
+   input [0:0] clk;
+   //----- GPOUT PORTS -----
+   output [0:19] gfpga_pad_ADDF_sumout;
+   
+The architecture will now look like :numref:`fig_addf_example`
+
+.. _fig_addf_example:
+
+.. figure:: ./figures/ADDF_Example_Spypad.svg

```

Comment:
The architecture shows a 20-bit  port of ``gfpga_pad_ADDF_sumout``. In the figure, you need to show the pin index correctly. What you did is to convert the ``sumout`` in each logic element to be spypads. You can explain in the caption of the figure that there are 10 logic elements in a CLB and you just show one of them (for the sake of illustration).

Also the caption of figures have to be reworked. This one should be something like ``An illustrative example of the sumout spypad sourced from an adder inside a logic element.
You may rework other captions.